crawler/processing/create_plate_exporter.py

```diff
@@ -135,4 +167,80 @@ def _record_source_plate_in_mongo_db(self, session: ClientSession) -> ExportResu
             )
 
     def _record_samples_in_mongo_db(self, session: ClientSession) -> ExportResult:
-        return ExportResult(success=True, create_plate_error=None)
+        message_uuid = self._message.message_uuid.value
+        LOGGER.debug(
+            f"Attempting to insert {self._message.total_samples} "
+            f"samples from message with UUID {message_uuid} into mongo..."
+        )
+
+        try:
+            try:
+                session_database = get_mongo_db(self._config, session.client)
+                samples_collection = get_mongo_collection(session_database, COLLECTION_SAMPLES)
+                result = samples_collection.insert_many(
+                    documents=self._mongo_sample_docs, ordered=False, session=session
+                )
+            except BulkWriteError as ex:
+                LOGGER.warning("BulkWriteError: will now establish whether this was because of duplicate samples.")
+
+                duplication_errors = list(
+                    filter(lambda x: x["code"] == 11000, ex.details["writeErrors"])  # type: ignore
+                )
+
+                if len(duplication_errors) == 0:
+                    # There weren't any duplication errors so this is not a problem with the message contents!
+                    raise
```

Comment:
Is this raising the same exception? How do we know what happened in this case?

Reply:
Yes it is the same exception.  You should never raise a new exception in an `except:` block as you lose the stack trace and other information.  By raising it again here, it gets caught in the outer `except:` block on line 213 so that it will be reported.  If that's not clear, we could duplicate lines `214 through 219` here and not re-raise, but it felt unnecessary to do that.

Reply:
No, just checking, I didnt know raise without arguments would reuse the same exception. That seems correct to me, to keep the same exception.

Reply:
Ah right yes.  You can only do it in an `except:` block.